test_conformance/computeinfo/extended_versioning.cpp

```diff
@@ -688,14 +698,41 @@ static int test_extended_versioning_device_built_in_kernels(cl_device_id device)
     return 0;
 }
 
+// Assumes the core enums, structures, and macros exactly match
+// the extension enums, structures, and macros:
+
+static_assert(CL_PLATFORM_NUMERIC_VERSION == CL_PLATFORM_NUMERIC_VERSION_KHR,
+              "CL_PLATFORM_NUMERIC_VERSION mismatch");
+static_assert(CL_PLATFORM_EXTENSIONS_WITH_VERSION
+                  == CL_PLATFORM_EXTENSIONS_WITH_VERSION_KHR,
+              "CL_PLATFORM_EXTENSIONS_WITH_VERSION mismatch");
+
+static_assert(CL_DEVICE_NUMERIC_VERSION == CL_DEVICE_NUMERIC_VERSION_KHR,
+              "CL_DEVICE_NUMERIC_VERSION mismatch");
+static_assert(CL_DEVICE_EXTENSIONS_WITH_VERSION
+                  == CL_DEVICE_EXTENSIONS_WITH_VERSION_KHR,
+              "CL_DEVICE_EXTENSIONS_WITH_VERSION mismatch");
+static_assert(CL_DEVICE_ILS_WITH_VERSION == CL_DEVICE_ILS_WITH_VERSION_KHR,
+              "CL_DEVICE_ILS_WITH_VERSION mismatch");
+static_assert(CL_DEVICE_BUILT_IN_KERNELS_WITH_VERSION
+                  == CL_DEVICE_BUILT_IN_KERNELS_WITH_VERSION_KHR,
+              "CL_DEVICE_BUILT_IN_KERNELS_WITH_VERSION mismatch");
+
+static_assert(sizeof(cl_name_version) == sizeof(cl_name_version_khr),
+              "cl_name_version mismatch");
+
+static_assert(CL_MAKE_VERSION(1, 2, 3) == CL_MAKE_VERSION_KHR(1, 2, 3),
+              "CL_MAKE_VERSION mismatch");
+
 int test_extended_versioning(cl_device_id deviceID, cl_context context,
                              cl_command_queue ignoreQueue, int num_elements)
 {
-    if (!is_extension_available(deviceID, "cl_khr_extended_versioning"))
+    int ext = is_extension_available(deviceID, "cl_khr_extended_versioning");
+    int core = get_device_cl_version(deviceID) >= Version(3, 0);
```

Comment:
bool

Reply:
Fixed.  FWIW, `is_extension_available` returns `int` not `bool`.  Is there any reason for this or should `is_extension_available` return `bool` also?

Reply:
Yeah, it should return `bool` really.